### PR TITLE
Release script to distinguish enforcer release and BOM release.

### LINKED ISSUE #878: Release script to distinguish BOM release and Enforcer Rule release
Release script to distinguish BOM release and Enforcer Rule release.

BOM release 1.0.0 should not block enforcer rule release 1.0.0 but they share the same name space in Git tag and branch name.

### PR DESCRIPTION
**Pull request by suztomo**
Fixes #878.

### CODE REVIEW

File: RELEASING.md [hunk 1]
```diff
@@ -4,7 +4,14 @@ To release artifacts 'dependencies-parent', 'dependencies', and 'linkage-checker
 follow the ["Developers/Releasing" steps in go/ct4j](
 https://g3doc.corp.google.com/company/teams/cloud-java/tools/developers/releasing.md?cl=head).
 
-Note: Cloud OSS BOM release procedure [boms/cloud-oss-bom/RELEASING.md](
+To distinguish the enforcer rule release and Google Cloud Libraries BOM release, the release script
+takes `enforcer` as the first argument:
```
Comment (elharo) [inline RELEASING.md]: Can't it also take bom? 
Should we have separate release docs for each product?

Comment (suztomo) [reply RELEASING.md]: The link to the BOM release doc is 5 lines below.

File: boms/cloud-oss-bom/RELEASING.md [hunk 2]
```diff
@@ -7,7 +7,7 @@ in `boms/cloud-oss-bom` directory:
 $ cd cloud-opensource-java/boms/cloud-oss-bom
 $ git checkout master
 $ git pull
-$ ../../scripts/prepare_release.sh 1.0.0 1.0.1
+$ ../../scripts/prepare_release.sh bom 1.0.0 1.0.1
```
Comment (suztomo) [inline boms/cloud-oss-bom/RELEASING.md]: This is release doc for BOM.

File: RELEASING.md (hunk 1 above)
Comment (elharo) [reply RELEASING.md]: This isn't clear. This is not just the enforcer rule release. It releases several artifacts.

Comment (suztomo) [reply RELEASING.md]: That's true. Let me think better name than `enforcer`.

**Comment by suztomo** [comment]
@elharo How about `dependencies` and `bom`?

File: RELEASING.md [hunk 3]
```diff
@@ -1,25 +1,30 @@
...
 
+Run the following Rapid release pipelines:
+
 | Artifact ID | Rapid project | Dependency |
 | ---------- | ------------------ | --------- |
 |dependencies-parent| [cloud-java-tools-cloud-opensource-java-parent-kokoro-release](http://rapid/cloud-java-tools-cloud-opensource-java-parent-kokoro-release)||
 |dependencies| [cloud-java-tools-cloud-opensource-java-dependencies-kokoro-release](http://rapid/cloud-java-tools-cloud-opensource-java-dependencies-kokoro-release)|dependencies-parent|
 |linkage-checker-enforcer-rules|[cloud-java-tools-cloud-opensource-java-enforcer-rules-kokoro-release](http://rapid/cloud-java-tools-cloud-opensource-java-enforcer-rules-kokoro-release)|dependencies-parent, dependencies|
 
-When releasing a version of `linkage-checker-enforcer-rules`, you need to release `dependencies`
```
Comment (suztomo) [inline RELEASING.md]: This sentences were unnecessarily unclear. As of now we release them altogether; there's no case where we only release `dependencies`.

**Pull request merged**